carml-engine/src/main/java/io/carml/engine/GetTemplateValue.java

```diff
@@ -45,22 +44,16 @@ public Optional<Object> apply(ExpressionEvaluation expressionEvaluation) {
   private void bindTemplateExpression(Expression expression, ExpressionEvaluation expressionEvaluation,
       Template.Builder templateBuilder) {
     templateBuilder.bind(expression, expr -> expressionEvaluation.apply(expr.getValue())
-        .map(this::prepareValueForTemplate));
+        .map(ExpressionEvaluation::extractValues)
```

Comment:
`.map(ExpressionEvaluation::extractValues)` returns an `Optional<List<String>>` on which `.map(this::prepareExtractedValuesForTemplate));` is called. So now with `ExpressionEvaluation::extractValues` it is assured that the list does not contain `null` values, correct?

Was the problem that `this::prepareExtractedValuesForTemplate` was passed a `List` containing a `null` value? Then `createNaturalRdfLexicalForm` was called with `null`?

Reply:
correct